### ISSUE TITLE
Redeclared names without usage rule

### ISSUE THREAD

**Issue by nikolaik** [pylint-dev/pylint#3471]
## Question
Is there an equivalent pylint rule to PyCharm's inspection _Redeclared names without usage_?

With the example code:
```python
class OptionType(graphene.InputObjectType):
    freeform_enabled = graphene.Boolean(required=True)
    freeform_enabled = GenericScalar(required=True)
```
PyCharm would give the warning:
![Screenshot from 2020-04-02 19-28-46](https://user-images.githubusercontent.com/104154/78279709-3fe72800-7518-11ea-93a1-f2dc50ea3fed.png)

**Comment by PCManticore** [comment]
There is not, but looks useful.

**Comment by neesara** [comment]
Hello, can some one please guide me on how to work on this issue?

**Comment by Pierre-Sassoulas** [comment]
Hello @neesara sorry for the time it took to reply. Did you check [the contributor documentation](https://pylint.pycqa.org/en/latest/development_guide/contributor_guide/contribute.html#tips-for-getting-started-with-pylint-development) ?

**Comment by DanielNoord** [comment]
I'm not sure if this should be in the same message, but flagging on the following pattern would be nice as well:

```python
lines = ["a"]
lines += ["b"]
```

We would need to think about:
```python
lines = ["a"]
line = get_lines()
lines += [line]
```

Don't think this should be part of initial implementation, but a generic name for this message would help.
